File: app/src/main/java/com/avoid/ihaveatheory/factory/ItemFactory.java
package com.avoid.ihaveatheory.factory;

import com.avoid.ihaveatheory.model.item.Arrow;
import com.avoid.ihaveatheory.model.item.Berries;
import com.avoid.ihaveatheory.model.item.Blanket;
import com.avoid.ihaveatheory.model.item.Bow;
import com.avoid.ihaveatheory.model.item.Hat;
import com.avoid.ihaveatheory.model.item.Hook;
import com.avoid.ihaveatheory.model.item.Item;
import com.avoid.ihaveatheory.model.item.ItemType;
import com.avoid.ihaveatheory.model.item.Meat;
import com.avoid.ihaveatheory.model.item.Pouch;
import com.avoid.ihaveatheory.model.item.Slippers;
import com.avoid.ihaveatheory.model.item.SmallPouch;
import com.avoid.ihaveatheory.model.item.Snare;
import com.avoid.ihaveatheory.model.item.WalkingStick;
import com.avoid.ihaveatheory.model.item.WaterBottle;
import com.avoid.ihaveatheory.model.item.Wood;

public class ItemFactory {
    public static Item getItem(ItemType itemType){
        Item item;
        switch (itemType){
            case BOW:
                item = new Bow();
                break;
            case ARROW:
                item = new Arrow();
                break;
            case SNARE:
                item = new Snare();
                break;
            case KNIFE:
                item = new Snare();
                break;
            case SLIPPERS:
                item = new Slippers();
                break;
            case HAT:
                item = new Hat();
                break;
            case WALKING_STICK:
                item = new WalkingStick();
                break;
            case SMALL_POUCH:
                item = new SmallPouch();
                break;
            case POUCH:
                item = new Pouch();
                break;
            case BLANKET:
                item = new Blanket();
                break;
            case HOOK:
                item = new Hook();
                break;
            case ROPE:
                item = new Hook();
                break;
            case WATER_BOTTLE:
                item = new WaterBottle();
                break;
            case BERRIES:
                item = new Berries();
                break;
            case WOOD:
                item = new Wood();
                break;
            case MEAT:
                item = new Meat();
                break;
            default:
                item = new Item() {};
        }

        return item;
    }
}

File: app/src/main/java/com/avoid/ihaveatheory/factory/ItemFactory.kt
package com.avoid.ihaveatheory.factory

import com.avoid.ihaveatheory.model.item.*

object ItemFactory {

    @JvmStatic
    fun getItem(itemType: ItemType): Item {
        var item: Item? = null
        when (itemType) {
            ItemType.BOW -> item = Bow()
            ItemType.ARROW -> item = Arrow()
            ItemType.SNARE -> item = Snare()
            ItemType.KNIFE -> item = Snare()
            ItemType.SLIPPERS -> item = Slippers()
            ItemType.HAT -> item = Hat()
            ItemType.WALKING_STICK -> item = WalkingStick()
            ItemType.SMALL_POUCH -> item = SmallPouch()
            ItemType.POUCH -> item = Pouch()
            ItemType.BLANKET -> item = Blanket()
            ItemType.HOOK -> item = Hook()
            ItemType.ROPE -> item = Hook()
            ItemType.WATER_BOTTLE -> item = WaterBottle()
            ItemType.BERRIES -> item = Berries()
            ItemType.WOOD -> item = Wood()
            ItemType.MEAT -> item = Meat()
        }

        return item
    }
}
